pkgs/build-support/ocaml/dune.nix

```diff
@@ -3,7 +3,15 @@
 { pname, version, nativeBuildInputs ? [], enableParallelBuilding ? true, ... }@args:
 
 let Dune =
-  let dune-version = args . duneVersion or (if args.useDune2 or true then "2" else "1"); in
+  let dune-version =
+    args.duneVersion or (
+        if builtins.hasAttr "useDune2" args then
+          if args.useDune2 or true then
+            builtins.trace "`useDune2 = true` is deprecated please consider removing it in ${pname}." builtins.trace "Dune is know to be retro-compatible, you can set duneVersion = \"2\" if breakage occur" "2"
+          else
+            builtins.trace "`useDune2 = false` is deprecated please consider replacing it in ${pname} with duneVersion = \"1\"" "1"
+        else "2"
+      ); in
```

Comment:
This is nonsense. Just use: `let dune-version = args . duneVersion or "3"; in`

Reply:
I thought on warn user a bit of time before actually removing. So you want I fail silently in case they did useDune2 = false because dune2 doesn't build ?